### PR TITLE
Automatically close the worker if the stream is garbage collected

### PR DESCRIPTION
**Pull request by mcollina**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -10,6 +10,25 @@ const {
...
+    return this._value
+  }
+}
+
+const FinalizationRegistry = global.FinalizationRegistry || class FakeFinalizationRegistry {
+  register () {}
+  unregister () {}
+}
+
+const registry = new FinalizationRegistry((worker) => {
```
Comment (jsumners) [inline index.js:28]: Does the `FakeFinalizationRegistry` need a constructor to handle this passed in anonymous function?

Comment (mcollina) [reply index.js:30]: nope he doesn't. It's just there so I can avoid a few ifs.

File: index.js [hunk 2]
```diff
@@ -10,6 +10,27 @@ const {
   READ_INDEX
 } = require('./lib/indexes')
 
+class FakeWeakRef {
```
Comment (davidmarkclements) [inline index.js:13]: should we do this or should we just drop support?

**Review by davidmarkclements — APPROVED** [review]
one question, otherwise looks great

File: index.js (hunk 2 above)
Comment (mcollina) [reply index.js:13]: We need to support Node.js v12, it's still LTS and we do not want to force one of our consumers (Fastify) to drop it.

**Review by jsumners — APPROVED** [review]
Looks good to me.

**Pull request merged**